bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,866 @@
+		replaceToggle.setVisible(enable);
+	}
+
+	private void enableReplaceTools(boolean enable) {
+		if (!okayToUse(replaceTools)) {
+			return;
+		}
+		((GridData) replaceTools.getLayoutData()).exclude = !enable;
+		replaceTools.setVisible(enable);
+
+		if (enable) {
+			((GridLayout) replaceTools.getParent().getLayout()).numColumns = 2;
+		} else {
+			((GridLayout) replaceTools.getParent().getLayout()).numColumns = 1;
+		}
+	}
+
+	private int getIdealDialogWidth(Rectangle targetBounds) {
+		int replaceToggleWidth = 0;
+		if (okayToUse(replaceToggle)) {
+			replaceToggleWidth = replaceToggle.getBounds().width;
+		}
+		int toolBarWidth = searchTools.getSize().x;
+		GC gc = new GC(searchBar);
+		gc.setFont(searchBar.getFont());
+		int idealWidth = gc.stringExtent(IDEAL_WIDTH_TEXT).x; // $NON-NLS-1$
+		int idealCompromiseWidth = gc.stringExtent(COMPROMISE_WIDTH_TEXT).x; // $NON-NLS-1$
+		int worstCompromiseWidth = gc.stringExtent(MINIMAL_WIDTH_TEXT).x; // $NON-NLS-1$
+		gc.dispose();
+
+		int newWidth = idealWidth + toolBarWidth + replaceToggleWidth;
+		if (newWidth > targetBounds.width * BIG_WIDTH_RATIO_EDITOR_TO_OVERLAY) {
+			newWidth = (int) (targetBounds.width * BIG_WIDTH_RATIO_EDITOR_TO_OVERLAY);
+			enableSearchTools(true);
+			enableReplaceTools(true);
+			enableReplaceToggle(true);
+		}
+		if (newWidth < idealCompromiseWidth + toolBarWidth) {
+			enableSearchTools(false);
+			enableReplaceTools(false);
+			enableReplaceToggle(true);
+		}
+		if (newWidth < worstCompromiseWidth + toolBarWidth) {
+			newWidth = (int) (targetBounds.width * WORST_CASE_RATIO_EDITOR_TO_OVERLAY);
+			enableReplaceToggle(false);
+			enableSearchTools(false);
+			enableReplaceTools(false);
+		}
+		return newWidth;
+	}
+
+	private Point getNewPosition(Widget targetTextWidget, Point targetOrigin, Rectangle targetBounds,
+			Point expectedSize) {
+		Point verticalScrollBarSize = ((Scrollable) targetTextWidget).getVerticalBar().getSize();
+		Point horizontalScrollBarSize = ((Scrollable) targetTextWidget).getHorizontalBar().getSize();
+
+		int newX = targetOrigin.x + targetBounds.width - expectedSize.x - verticalScrollBarSize.x
+				- ((StyledText) targetTextWidget).getRightMargin();
+		int newY = targetOrigin.y;
+		if (!positionAtTop) {
+			newY += targetBounds.height - expectedSize.y - horizontalScrollBarSize.y;
+		}
+		return new Point(newX, newY);
+	}
+
+	/**
+	 * When making the text-bar 100% small and then regrowing it, we want the text
+	 * to start at the first character again.
+	 */
+	private void repositionTextSelection() {
+		if (okayToUse(searchBar) && !searchBar.isFocusControl()) {
+			searchBar.setSelection(0, 0);
+		}
+		if (okayToUse(replaceBar) && !replaceBar.isFocusControl()) {
+			replaceBar.setSelection(0, 0);
+		}
+	}
+
+	private void positionToPart() {
+		getShell().requestLayout();
+		if (!(targetPart instanceof StatusTextEditor)) {
+			return;
+		}
+
+		StatusTextEditor textEditor = (StatusTextEditor) targetPart;
```

Comment:
This could use pattern-matching for `instanceof`
```suggestion
		if (!(targetPart instanceof StatusTextEditor textEditor)) {
			return;
		}
```

Reply:
I dissagree (and I specifically wrote this code thinking of the option you proposed)! The negate would need to have the entire rest of the function as scope, leading to increased complexity of the program, leading to a higher [cyclomatic complexity](https://codinghelmet.com/articles/reduce-cyclomatic-complexity-guard-clause)